pkg/template/api/types.go

```diff
@@ -75,3 +75,82 @@ type Parameter struct {
 	// Optional: Indicates the parameter must have a value.  Defaults to false.
 	Required bool
 }
+
+// TemplateInstance is part of an experimental API.
+type TemplateInstance struct {
+	unversioned.TypeMeta
+	kapi.ObjectMeta
+
+	Spec   TemplateInstanceSpec
+	Status TemplateInstanceStatus
+}
+
+type TemplateInstanceSpec struct {
+	Template        Template
+	SecretRef       kapi.LocalObjectReference
+	ImpersonateUser string
+}
+
+type TemplateInstanceStatus struct {
+	Conditions []TemplateInstanceCondition
+}
+
+type TemplateInstanceCondition struct {
+	Type               TemplateInstanceConditionType
+	Status             kapi.ConditionStatus
+	LastTransitionTime unversioned.Time
+	Reason             string
+	Message            string
+}
+
+type TemplateInstanceConditionType string
+
+const (
+	TemplateInstanceCreated TemplateInstanceConditionType = "Created"
+	TemplateInstanceFailed  TemplateInstanceConditionType = "Failed"
+)
+
+type TemplateInstanceList struct {
+	unversioned.TypeMeta
+	unversioned.ListMeta
+
+	Items []TemplateInstance
+}
+
+// TemplateBrokerUUIDMap is part of an experimental API.
+type TemplateBrokerUUIDMap struct {
```

Comment:
The object stores backing state for the template implementation of the servicebroker API.  Each object instance maps an externally provided provision (UU)ID to a namespaced TemplateInstance object.  When servicebroker clients create bindings against a provision (UU)ID, these are recorded in the object as well.